src/services/db/GitFileSystemService.ts

```diff
@@ -433,9 +430,98 @@ export default class GitFileSystemService {
     })
   }
 
-  // TODO: Creates either directory or file
-  // ResourceDirectoryService used this to create a directory + file at the same time
-  create() {}
+  // Creates a file and the associated directory if it doesn't exist
+  create(
+    repoName: string,
+    userId: string,
+    content: string,
+    directoryName: string,
+    fileName: string
+  ): ResultAsync<
+    GitCommitResult,
+    ConflictError | GitFileSystemError | NotFoundError
+  > {
+    const filePath = directoryName ? `${directoryName}/${fileName}` : fileName
+    const pathToEfsDir = `${EFS_VOL_PATH}/${repoName}/${directoryName}/`
+    const pathToEfsFile = `${EFS_VOL_PATH}/${repoName}/${filePath}`
+    const encodedContent = content
+    let oldStateSha = ""
+
+    return this.getLatestCommitOfBranch(repoName, BRANCH_REF)
+      .andThen((latestCommit) => {
+        const { sha } = latestCommit
+        if (!sha) {
+          return errAsync(new GitFileSystemError("An unknown error occurred"))
+        }
+        oldStateSha = sha
+        return okAsync(true)
+      })
+      .andThen(() => this.getFilePathStats(repoName, directoryName))
+      .andThen(() => ok(""))
+      .orElse((error) => {
+        if (error instanceof NotFoundError) {
+          // Create directory if it does not already exist
+          try {
+            fs.promises.mkdir(pathToEfsDir)
+            return ok("")
+          } catch (mkdirErr) {
+            logger.error(
+              `Error occurred while creating ${pathToEfsDir} directory: ${mkdirErr}`
+            )
+            return err(new GitFileSystemError("An unknown error occurred"))
+          }
+        }
+        return err(error)
+      })
+      .andThen(() => this.getFilePathStats(repoName, filePath))
+      .andThen((stats) => {
+        if (stats.isFile())
+          return err(
+            new ConflictError(
+              `File ${filePath} already exists in repo ${repoName}`
+            )
+          )
+        return ok("")
+      })
+      .orElse((error) => {
+        if (error instanceof NotFoundError) {
```

Comment:
where do we get the NotFoundError from?

Reply:
This is from `this.getFilePathStats(repoName, filePath)` - if it's not found, the file does not yet exist